Sources/Rego/IREvaluator.swift

```diff
@@ -2,6 +2,119 @@ import AST
 import Foundation
 import IR
 
+// Custom array type that supports IR.Local indexing
+internal struct Locals: Equatable, Sendable, Encodable {
```

Comment:
⛏️ I'd suggest moving this type lower down in the file or to its own `Locals.swift` file. It's a bit out of context before introducing the `IREvaluator` and other related types.